winch/codegen/src/codegen/mod.rs

```diff
@@ -948,9 +948,9 @@ where
             &mut self.context,
             Callee::Builtin(out_of_fuel.clone()),
         );
+        self.raise_if_host_trapped();
         // Under fuel limits branch.
         self.masm.bind(continuation);
-        self.context.free_reg(fuel_var);
```

Comment:
Similarly @saulecabrera I had to move this up since `fuel_var` was assigned rax but rax was needed as the return value of the hostcall, so this was needed to fix some panics during compilation.